src/storage/SimpleResourceStore.ts

```diff
@@ -11,43 +11,89 @@ import streamifyArray from 'streamify-array';
 import { StreamWriter } from 'n3';
 import { UnsupportedMediaTypeHttpError } from '../util/errors/UnsupportedMediaTypeHttpError';
 
+/**
+ * Resource store storing its data as Quads in an in-memory map.
+ * All requests will throw an {@link NotFoundHttpError} if unknown identifiers get passed.
+ */
 export class SimpleResourceStore implements ResourceStore {
   private readonly store: { [id: string]: Quad[] } = { '': []};
   private readonly base: string;
   private index = 0;
 
+  /**
+   * @param base - Will be stripped of all incoming URIs and added to all outgoing ones to find the relative path.
+   */
   public constructor(base: string) {
     this.base = base;
   }
 
+  /**
+   * Stores the incoming data under a new URL corresponding to `container.path + number`.
+   * Slash added when needed.
+   * @param container - The identifier to store the new data under.
+   * @param representation - Data to store. Only Quad streams are supported.
+   *
+   * @returns The newly generated identifier.
+   */
   public async addResource(container: ResourceIdentifier, representation: Representation): Promise<ResourceIdentifier> {
     const containerPath = this.parseIdentifier(container);
-    const newPath = `${containerPath}/${this.index}`;
+    const newPath = `${containerPath}${container.path.endsWith('/') ? '' : '/'}${this.index}`;
```

Comment:
This is where I usually `containerPath.replace(/\/*$/, '/'_`.
Or given that we will be using this often, an `ensureTrailingSlash` function exported from some utility.